### ISSUE TITLE
Dart SDK location not found on VS Code/Pixelbook/Crostini

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#1013]
Moving from https://github.com/flutter/flutter/issues/18502, opened by @aeroperf.

## Steps to Reproduce

On a Pixelbook running Linux in Crostini. I open VS Code and it loads. After about 15-20 seconds, an analysis starts then the PROBLEMS tab fills with errors (10k+). Flutter and other subfolders on the left pane turn red. VS Code often locks up during this analysis.

I know this works as I was able to get everything working previously (but then restarted a Crostini container for other reasons and lost Flutter). When it did work, I did not have to install a Dart SDK or specify its location, it just worked. I verified that the user settings path to Dart includes the sdk, and it does.

I do not have Android Studio installed and wonder if that's somehow the problem, but can't see why that would be the case. 

I also noticed flutter doctor indicated the VS Code Dart extension is not installed, even though within VS Code, it shows it as installed.

1. Follow instructions in [this post](https://medium.com/@timsneath/flutter-development-on-a-pixelbook-dde984a3fc1e) to get and install Flutter and VS Code
2. Open VS Code; after a few seconds, it find 10k+ errors and the flutter folder is highlighted red.
3. In the past I was given the option to located Dart-SDK location, and when clicking on it, nothing happened.
4. More recently, I'm not given that option anymore.

## Logs

```
User settings:
"dart.sdkPath": "/home/mrv/flutter/bin/cache/dart-sdk/bin",
"dart.flutterSdkPath": "/home/mrv/flutter"
```

```
[✓] Flutter (Channel beta, v0.4.4, on Linux, locale en_US.UTF-8)
    • Flutter version 0.4.4 at /home/mrv/flutter
    • Framework revision f9bb4289e9 (5 weeks ago), 2018-05-11 21:44:54 -0700
    • Engine revision 06afdfe54e
    • Dart version 2.0.0-dev.54.0.flutter-46ab040e58

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.

[✗] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[!] VS Code (version 1.24.1)
    • VS Code at /usr/share/code
    • Dart Code extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.dart-code

[!] Connected devices
    ! No devices available

! Doctor found issues in 4 categories.
```

**Comment by DanTup** [comment]
Running on a Pixelbook isn't something I've tried so you're a little in uncharted territory. I know others have had success though, so hopefully it's nothing serious.

What sort of errors are being reported in the problems view? Your `doctor` output suggests you don't have `Dart Code` installed; but I'm wondering if that's down to the setup you have (running on a Pixelbook isn't something I've tested!) since it shouldn't be able to generate errors if you don't have it installed!

Could you try enabling the [Analysis Server](https://dartcode.org/docs/logging/#analyzer) logging and restart Code and get that log file? Once you have it (please review it for sensitive content; it may include content from your open files) please attach here or email to logs@dartcode.org and we'll go from there.

**Comment by timsneath** [comment]
As noted in the blog post, running Flutter on a Pixelbook is at best 'experimental', so no promises from our side. Android Studio is a dependency because of the build tools, so it may well be that this is your problem. Is there a reason why you're not installing it?

**Comment by aeroperf** [comment]
That's most likely it @timsneath . I had it running once without issues, but had AS installed. I destroyed that container for several reasons and decided not to install AS on the second go, since it wasn't clear to me it was a dependency. I'll add it back in and report back, but I'm guessing that is it. Thanks for pointing this out.

**Comment by DanTup** [comment]
Let me know how you get on. Missing AS shouldn't result in Dart SDK location issues, or in analysis errors reported; so there may be something else wrong.

I think ensuring Dart Code works correctly is probably not difficult (there aren't a lot of moving parts); but whether you'll get any further (eg being able to run/debug flutter) I'm not sure.

The error above about not having Dart Code installed in `doctor` is a bug I've got [a PR](https://github.com/flutter/flutter/pull/18515) open to fix, but it won't affect anything else. If you can supply the info requested above (logs, copy of the sorts of things being reported in the problems pane) it might be clearer what's going on.

My guess is the reason you're not prompted to pick an SDK is because one was found (if it wasn't, it's unlikely you'd see any errors reported).

**Comment by aeroperf** [comment]
@DanTup  AS install did not fix it. I captured a log file and emailed it, referencing this case.

**Comment by DanTup** [comment]
@aeroperf What folder(s) are you opening? In the analysis results I see things like:

- /home/mrvogt73/flutter/bin/cache/dart-sdk/lib/_http/websocket.dart

This really should not be being analyzed! Could you capture a log again using [dart.analyzerLogFile](https://dartcode.org/docs/logging/#analyzer) and then re-open the project? That way the log will contain the `setAnalysisRoots` call which might explain what's going wrong.

**Comment by aeroperf** [comment]
I just started "Capture Logs" at VS Code launch. How/where do I set the file path for dart.analyzerLogFile, as per instructions, in VS Code?

**Comment by DanTup** [comment]
@aeroperf You set this in the VS Code User Settings (run the `Open User Settings` command). Be sure to remove it afterwards to avoid it logging forever. I'll make a note to make the website clearer how to edit the settings, thanks!

**Comment by DanTup** [comment]
Ok, so in the log there is this:

```js
{"id":"2","method":"analysis.setAnalysisRoots","params":{"excluded":[],"included":["/home/mrvogt73"]}}
```

It seems like you have your entire home directory open. I wonder whether this is why the server is scanning the entire tree (I would normally expect it to skip the `bin/cache` folder from flutter).

Can you try opening just `/home/mrvogt73/flutter` and see if that makes a difference?

**Comment by DanTup** [comment]
Actually, is there a reason you're opening the flutter folder at all? If you're not working on Flutter, you don't need to have your flutter install (`/home/mrvogt73/flutter/`) open at all.

**Comment by aeroperf** [comment]
Hmmm. Good point on the folder open. I closed it (not realizing that VS Code operated this way) and no issues. thanks for pointing this out. Guess I'd better stick with the more familiar Android Studio here. This case can be closed.

**Comment by DanTup** [comment]
@aeroperf It shouldn't cause issues having it open, but it will mean the entire tree is being analyzed. Opening the flutter folder directly doesn't cause any issues for me, so I wonder if the server does something to skip over the `bin/cache` folder normally but maybe doesn't do it here (@bwilkerson - do you know how that works? is there special handling for the flutter repo that might not trigger if you open its parent?).

Also; if you do want the flutter code open, you can use the multi-root support in VS Code to have multiple folders open. It works best if you have each "project" as a root though, so we know where to launch when you hit `F5`.

> Guess I'd better stick with the more familiar Android Studio here

That sounds a bit drastic :) I'm happy to assist if I can if you're having issues with (or unsure how to use) VS Code / Dart Code. And if we can make any changes to make things smoother for new users, I'm all ears!

**Comment by aeroperf** [comment]
I think in the end, this can all be contributed to user error (apologies!). While there might be an underlying scanning issue, I failed to fully understand VS Code's open folder workings. somehow, I managed to open the root folder after install everything was open, including the hidden folders. Once I closed that and only opened my Projects folder, the issue disappeared. 

TL;DR, I need to read up more on VS Code...
Thank you for your help an patience!
